bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/TitleImplV2.java

```diff
@@ -0,0 +1,148 @@
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { FormTitle.class, ComponentExporter.class },
+    resourceType = FormConstants.RT_FD_FORM_TITLE_V2)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+
+public class TitleImplV2 extends AbstractFormComponentImpl implements FormTitle {
+
+    /**
+     * The current resource.
+     */
+    @SlingObject
+    private Resource resource;
+
+    @ScriptVariable(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @JsonIgnore
+    @Nullable
+    private Style currentStyle;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Nullable
+    private String type;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL, name = JcrConstants.JCR_TITLE)
+    @Nullable
+    private String title;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL, name = "format")
+    @Nullable
+    private String format;
+
+    private Heading heading;
+
+    /**
+     * Translation of the title property
+     */
+
+    @PostConstruct
+    private void initModel() {
+        if (request != null && i18n == null) {
+            i18n = GuideUtils.getI18n(request, resource);
+        }
+        if (StringUtils.isBlank(title)) {
+            Resource formContainerResource = ComponentUtils.getFormContainer(resource);
+            if (formContainerResource != null) {
+                title = formContainerResource.getValueMap().get("title", String.class);
+            }
+        }
+        if (heading == null) {
+            heading = Heading.getHeading(format);
+            if (heading == null && currentStyle != null) {
+                heading = Heading.getHeading(currentStyle.get(PN_DESIGN_DEFAULT_FORMAT, String.class));
+            }
+        }
+    }
+
+    @JsonIgnore
+    @Override
+    public String getText() {
+        return getValue();
+    }
+
+    @Override
+    public String getValue() {
+        String propertyName = JcrConstants.JCR_TITLE;
+        String propertyValue = title;
+        boolean editMode = true;
+        if (request != null) {
+            editMode = WCMMode.fromRequest(request) == WCMMode.EDIT || WCMMode.fromRequest(request) == WCMMode.DESIGN;
+        }
+        if (editMode) {
+            return propertyValue;
+        }
+        if (StringUtils.isBlank(propertyValue)) {
+            return null;
+        }
+        return ComponentUtils.translate(propertyValue, propertyName, resource, i18n);
+    }
+
+    @JsonIgnore
+    public String getFormat() {
+        if (heading != null) {
+            return heading.getElement();
+        }
+        return null;
+    }
+
+    @NotNull
+    protected FormComponentData getComponentData() {
+        return super.getComponentData();
+    }
+
+    @Override
+    public String getName() {
+        return null;
```

Comment:
Name is mandatory if a component is a form component, please check other implementation. We should also expose this in dialog

Reply:
Implemented!

Reply:
Why the title component should have a name. We have the chance to enable unnamed fields now, though we do not need to expose it but we can fix in our code wherever we have unnamed fields.

Reply:
> Why the title component should have a name. We have the chance to enable unnamed fields now, though we do not need to expose it but we can fix in our code wherever we have a hard restriction on name property

In the core components, the name is mandatory as per the initial design, even though af-core does not enforce this requirement. While we can remove the name from the title component, doing so would prevent dynamic changes to the title component's text. Currently, we don't have a use case for this dynamic capability, so I'm okay with not implementing it.